csrc/kernel_cache.cpp

```diff
@@ -481,23 +481,19 @@ FusionKernelRuntime::FusionKernelRuntime(
   // Initialize the evaluator simplifer
   precomputed_values_ = std::make_unique<PrecomputedValues>(fusion.get());
 
-  //! Try to schedule the complete fusion
-  scheduler_debug_utils::canScheduleMessage(
-      "***Runtime***: Try to schedule fusion un-segmented:\n");
-
-  const auto maybe_complete_fusion_heuristic =
-      SchedulerEntry::proposeHeuristics(fusion.get(), runtime_info);
-
-  //! Decide if this fusion is segmented or not
-  const bool segmented = !maybe_complete_fusion_heuristic.has_value();
-
-  if (segmented) {
-    // Take ownership and segment transformed fusion
+  if (!SegmentCandidateFinder::hasSegmentHints(fusion.get())) {
```

Comment:
nope... If you look at the logic above. executor asks about `scheduler` first, before even consulting segmenter.
So if any single scheduler can handle the fusion in a whole, segmenter is totally skipped.

Reply:
Oh, I see. Then, shouldn't we change `fromCompleteFusion` so that it fails when there's a segmentaion hint? It seems to me a little inconsistent that `fromCompleteFusion` can succeed even with a segmentation hint.

Reply:
Good point. I can add a check there.
It feels like we can go one step further and clean up the API here. `SegmentedFusion::fromCompleteFusion` and `SegmentCandidateFinder::segment` really should have been one function instead. I'll update the API.